storage/mysql/mysql.go

```diff
@@ -0,0 +1,653 @@
+		model, err := m.ReadAuthorizationModel(ctx, store, modelIDs[i])
+		if err != nil {
+			return nil, nil, err
+		}
+		models = append(models, model)
+	}
+
+	return models, token, nil
+}
+
+func (m *MySQL) FindLatestAuthorizationModelID(ctx context.Context, store string) (string, error) {
+	ctx, span := m.tracer.Start(ctx, "mysql.FindLatestAuthorizationModelID")
+	defer span.End()
+
+	var modelID string
+	stmt := "SELECT authorization_model_id FROM authorization_model WHERE store = ? ORDER BY authorization_model_id DESC LIMIT 1"
+	err := m.db.QueryRowContext(ctx, stmt, store).Scan(&modelID)
+	if err != nil {
+		return "", handleMySQLError(err)
+	}
+
+	return modelID, nil
+}
+
+func (m *MySQL) ReadTypeDefinition(
+	ctx context.Context,
+	store, modelID, objectType string,
+) (*openfgapb.TypeDefinition, error) {
+	ctx, span := m.tracer.Start(ctx, "mysql.ReadTypeDefinition")
+	defer span.End()
+
+	var marshalledTypeDef []byte
+	stmt := "SELECT type_definition FROM authorization_model WHERE store = ? AND authorization_model_id = ? AND type = ?"
+	err := m.db.QueryRowContext(ctx, stmt, store, modelID, objectType).Scan(&marshalledTypeDef)
+	if err != nil {
+		return nil, handleMySQLError(err)
+	}
+
+	var typeDef openfgapb.TypeDefinition
+	if err := proto.Unmarshal(marshalledTypeDef, &typeDef); err != nil {
+		return nil, err
+	}
+
+	return &typeDef, nil
+}
+
+func (m *MySQL) MaxTypesInTypeDefinition() int {
+	return m.maxTypesInTypeDefinition
+}
+
+func (m *MySQL) WriteAuthorizationModel(
+	ctx context.Context,
+	store, modelID string,
+	tds *openfgapb.TypeDefinitions,
+) error {
+	ctx, span := m.tracer.Start(ctx, "mysql.WriteAuthorizationModel")
+	defer span.End()
+
+	if len(tds.GetTypeDefinitions()) > m.MaxTypesInTypeDefinition() {
+		return storage.ExceededMaxTypeDefinitionsLimitError(m.maxTypesInTypeDefinition)
+	}
+
+	stmt := "INSERT INTO authorization_model (store, authorization_model_id, type, type_definition) VALUES (?, ?, ?, ?)"
+
+	tx, err := m.db.BeginTx(ctx, nil)
+	if err != nil {
+		return handleMySQLError(err)
+	}
+	defer rollbackTx(ctx, tx, m.logger)
+
+	for _, typeDef := range tds.GetTypeDefinitions() {
+		marshalledTypeDef, err := proto.Marshal(typeDef)
+		if err != nil {
+			return err
+		}
+
+		_, err = tx.ExecContext(ctx, stmt, store, modelID, typeDef.GetType(), marshalledTypeDef)
+		if err != nil {
+			return handleMySQLError(err)
+		}
+	}
+
+	err = tx.Commit()
+	if err != nil {
+		return handleMySQLError(err)
+	}
+
+	return nil
+}
+
+func (m *MySQL) CreateStore(ctx context.Context, store *openfgapb.Store) (*openfgapb.Store, error) {
+	ctx, span := m.tracer.Start(ctx, "mysql.CreateStore")
+	defer span.End()
+
+	var createdAt time.Time
+	stmt := "INSERT INTO store (id, name, created_at, updated_at) VALUES (?, ?, NOW(), NOW())"
+	_, err := m.db.ExecContext(ctx, stmt, store.Id, store.Name)
+	if err != nil {
+		return nil, handleMySQLError(err)
+	}
+
+	var id, name string
+	stmt = "SELECT id, name, created_at FROM store WHERE id = ?"
```

Comment:
Oh yes, I meant it is good to have so that the implementation are as close as possible. I was just wondering if MySQL had a RETURNING statement, which some searching tells me it doesn't, but MariaDB does https://mariadb.com/kb/en/insertreturning/. 

Now I am wondering if these two statements should be in a transaction. In case the second statement fails after the first we will return an error to the user, but actually the data will exists in MySQL. WDYT?